tests/test_rio_info.py

```diff
@@ -8,6 +8,361 @@
 from rasterio.rio.main import main_group
 
 
+with rasterio.Env() as env:
+    HAVE_NETCDF = 'NetCDF' in env.drivers().keys()
+
+
+def test_delete_nodata_exclusive_opts(data):
+    """--unset-nodata and --nodata can't be used together"""
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--unset-nodata', '--nodata', '0'])
+    assert result.exit_code == 2
+
+
+def test_delete_crs_exclusive_opts(data):
+    """--unset-crs and --crs can't be used together"""
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--unset-crs', '--crs', 'epsg:4326'])
+    assert result.exit_code == 2
+
+
+@pytest.mark.skip(
+    parse(rasterio.__gdal_version__) < parse('1.10'),
+    reason='GDAL version >= 1.10 required')
+def test_unset_crs(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(main_group,
+                           ['edit-info', inputfile, '--unset-crs'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.crs is None
+
+
+@pytest.mark.skip(
+    parse(rasterio.__gdal_version__) >= parse('1.10'),
+    reason='Test applies to GDAL version < 1.10')
+def test_unset_crs_gdal19(data):
+    """unsetting crs doesn't work for geotiff and gdal 1.9
+    and should emit an warning"""
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    with rasterio.open(inputfile) as src:
+        orig_crs = src.crs
+    with pytest.warns(UserWarning):
+        result = runner.invoke(main_group,
+                               ['edit-info', inputfile, '--unset-crs'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.crs == orig_crs  # nochange
+
+
+def test_edit_nodata_err(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(main_group,
+                           ['edit-info', inputfile, '--nodata', '-1'])
+    assert result.exit_code == 2
+
+
+@pytest.mark.xfail(
+    Version(rasterio.__gdal_version__) < Version('2.1'),
+    reason='GDAL version >= 2.1 required')
+def test_delete_nodata(data):
+    """Delete a dataset's nodata value"""
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--unset-nodata'])
+    assert result.exit_code == 0
```

Comment:
In addition to making sure it ran successfully, it would probably be good to check that the output has no metadata value.

Reply:
This only shows up as new due to the test shuffle but I added a check.